src/libstd/io/mod.rs

```diff
@@ -989,6 +1040,56 @@ impl<'a> IoSlice<'a> {
     pub fn new(buf: &'a [u8]) -> IoSlice<'a> {
         IoSlice(sys::io::IoSlice::new(buf))
     }
+
+    /// Advance the internal cursor of the slice.
+    ///
+    /// # Notes
+    ///
+    /// Elements in the slice may be modified.
+    ///
+    /// # Examples
```

Comment:
The documentation here I think may also want to mention that this API panics if you advance beyond the cumulative size of all buffers.

Reply:
But it doesn't panic if you go beyond the cumulative size, see the `io_slice_advance_beyond_total_length` test.